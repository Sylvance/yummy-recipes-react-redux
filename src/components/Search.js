import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';

export default class Search extends Component {
  static propTypes = {
    onSearch: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired,
    forComponent: PropTypes.string.isRequired
  }

  state = {
    title: this.props.title || ''
  }

  handleSearch = (e) => {
      e.preventDefault(); 

      const { name, value } = e.target;
      this.setState({ [name]: value })
      setTimeout(this.handleSave, 300)
  }

  handleSave = () => {
    let url = null
      let theComponent = this.props.forComponent
      if (theComponent === 'category') {
        url = `/api/users/${this.props.id}/categories?title=${this.state.title}`
      } else {
        url = `/api/categories/${this.props.id}/recipes?title=${this.state.title}`
      }
      this.props.onSearch(url)
  }

  render() {
    return (
      <Form name="form">
        <FormGroup>
          <Label for="title">Search by title</Label>
          <Input type="text"
              placeholder="Title"
              autoFocus="true"
              name="title"
              onChange={this.handleSearch}
              required/>
        </FormGroup>
      </Form>
    )
  }
}
