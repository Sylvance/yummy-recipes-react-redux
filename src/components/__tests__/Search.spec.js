// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('Search test', () => {
  it('Search should match snapshot', () => {
    const component = renderer.create(
      <Search onSearch={() => {}} id={42} forComponent={'defaultString'} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
