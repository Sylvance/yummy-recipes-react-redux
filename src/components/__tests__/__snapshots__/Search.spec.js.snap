// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search test Search should match snapshot 1`] = `
<form
  className=""
  name="form"
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="title"
    >
      Search by title
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="title"
      onChange={[Function]}
      placeholder="Title"
      required={true}
      type="text"
    />
  </div>
</form>
`;
